pkg/lint/lint_test.go

```diff
@@ -205,10 +227,28 @@ func TestGoModLint(t *testing.T) {
 	}
 
 	for _, modFile := range modFiles {
+		assert.Equalf(t, runtime.Version()[2:], modFile.Go.Version, "%s's go directive should be match runtime.Version(): %q", modFile.Module.Mod.Path, runtime.Version()[2:])
+
 		for _, r := range modFile.Require {
 			if strings.HasPrefix(r.Mod.Path, modPrefix) {
 				assertPresentInMainOrRelease(t, modFile, r.Mod)
 			}
+
+			if version, ok := workspaceWideVersions[r.Mod.Path]; ok {
+				assert.Equalf(t, version, r.Mod.Version, `
+%s MUST be %s across the all go.mod's in this repository.
+
+	%s is using %s
+
+Quick Fixes:
+
+	go get %s@%s
+	go mod edit -require=%s@%s
+	%s %s
```

Comment:
Per https://github.com/redpanda-data/redpanda-operator/pull/419#discussion_r1931932259 I'm not sure what the output would mean.